Before loading data. it was necesarry to re_encode the data.  Foreign characters were not encoded correctly.  Please see re_encode.py for this implementation.

In [4]:
import ReadData;
from copy import deepcopy;
from igraph import *
from scipy import stats
import numpy as np

We read the data into memory using handy functions developed by Yibo Yao.  Please see: http://www.eecs.wsu.edu/~yyao/DirectedStudyI/Datasets/MovieLens/ReadData.py

The source for this function is included in my repo for convenience only.  Full credit to him for its development.

In [5]:
import ReadData;
ReadData.readUserInfo();
ReadData.readMovieInfo();
ReadData.readRatingInfo();
dates = ReadData.dateDict;
dateLi = dates['2000-04-25'];

In [14]:
import WriteData;
# call custom developed function that will build an iGraph object for us
g = WriteData.generateDailyGraph(dateLi)
summary(g)

IGRAPH UN-T 184 376 -- 
+ attr: age (v), gender (v), genre (v), name (v), title (v), type (v)


In [13]:
# Calculate some metrics, including diameter (required by assignment) for entire graph
diameter = Graph.diameter(g)
radius = Graph.radius(g)
ec = Graph.eccentricity(g)

eigen = Graph.eigenvector_centrality(g)
betweenness = Graph.betweenness(g)
closeness = Graph.closeness(g)
degree = Graph.degree(g)

print "Overall Graph Analysis"
print "Overall Degree: " + str(degree)
print "Overall Eigen vector centrality: " + str(eigen)
print "Overall Betweenness: " + str(betweenness)
print "Overall Closeness: " + str(closeness)

Overall Graph Analysis
Overall Degree: [12, 2, 2, 2, 4, 2, 4, 234, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 4, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 4, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 4, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 4, 2, 2, 2, 2, 2, 2, 2, 2, 4, 2, 2, 2, 2, 2, 2, 130, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2]
Overall Eigen vector centrality: [0.017935297086283677, 0.0016545029787667289, 0.0016545029787667289, 0.00165450297876673, 0.09390293191792347, 0.0016545029787667263, 0.09390293191792344, 1.0, 0.09224842893915673, 0.09224842893915675, 0.09224842893915675, 0.09224842893915673, 0.09224842893915675, 0.09224842893915675, 0.09224842893915673, 0.09224842893915675, 0.09224842893915675, 0.09224842893915675, 

In [8]:
# Calculate some comedy specific metrics
comedyVs = g.vs.select(lambda x:x["type"]=="movie" and x["genre"]=="Comedy")

print "Comedy Movie Analyis:"
print "Comedy Degree: " + str(g.degree(comedyVs))
print "Comedy Eigen vector centrality: " + str(g.eigenvector_centrality(comedyVs))
print "Comedy Betweenness: " + str(g.betweenness(comedyVs))
print "Comedy Closeness: " + str(g.closeness(comedyVs))

Comedy Movie Analyis:
Comedy Degree: [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 4, 2, 2, 2]
Comedy Eigen vector centrality: [0.0179352970862836, 0.0016545029787666946, 0.0016545029787666933, 0.001654502978766692, 0.09390293191792341, 0.0016545029787666946, 0.09390293191792343, 1.0, 0.09224842893915676, 0.09224842893915676, 0.09224842893915672, 0.09224842893915679, 0.09224842893915675, 0.09224842893915676, 0.09224842893915676, 0.09224842893915672, 0.09224842893915676, 0.09224842893915672, 0.09224842893915676, 0.09224842893915675, 0.09224842893915676, 0.09224842893915676, 0.09224842893915676, 0.09224842893915675, 0.09224842893915672, 0.09224842893915675, 0.09224842893915676, 0.09224842893915672, 0.09224842893915676, 0.09224842893915675, 0.09224842893915675, 0.09224842893915676, 0.10103199384485254, 0.09224842893915672, 0.09224842893915672, 0.09224842893915676, 0.09224842893915672, 0.09224842893915672, 0.09224842893915672, 0.09224842893915675, 0.09224842893915676, 0.09224842893915676, 0.0922484289

In [9]:
# Calculate some drama specific metrics
dramaVs = g.vs.select(lambda x: x["type"] == "movie" and x["genre"] == "Documentary")
print "Drama Movie Analyis:"
print "Drama Degree: " + str(g.degree(dramaVs))
print "Drama Eigen vector centrality: " + str(g.eigenvector_centrality(dramaVs))
print "Drama Betweenness: " + str(g.betweenness(dramaVs))
print "Drama Closeness: " + str(g.closeness(dramaVs))

Drama Movie Analyis:
Drama Degree: [2, 2, 2, 2, 2, 2]
Drama Eigen vector centrality: [0.01793529708628359, 0.0016545029787666846, 0.001654502978766692, 0.001654502978766692, 0.09390293191792341, 0.001654502978766691, 0.09390293191792347, 1.0, 0.0922484289391568, 0.09224842893915679, 0.0922484289391568, 0.0922484289391568, 0.0922484289391568, 0.09224842893915679, 0.09224842893915679, 0.09224842893915679, 0.09224842893915679, 0.09224842893915679, 0.09224842893915676, 0.0922484289391568, 0.09224842893915676, 0.09224842893915676, 0.09224842893915676, 0.09224842893915676, 0.09224842893915676, 0.09224842893915676, 0.09224842893915679, 0.09224842893915679, 0.09224842893915679, 0.09224842893915679, 0.09224842893915679, 0.09224842893915675, 0.10103199384485269, 0.09224842893915679, 0.09224842893915676, 0.09224842893915676, 0.09224842893915679, 0.09224842893915679, 0.0922484289391568, 0.09224842893915676, 0.09224842893915676, 0.09224842893915679, 0.09224842893915679, 0.09224842893915676, 0.09224

In [12]:
# Perform T test between comedy and drama metrics
t, p =  stats.ttest_ind(np.array(g.betweenness(comedyVs)), np.array(g.betweenness(dramaVs)), equal_var=False)
print t
print p

1.0
0.334281943395


With p = 0.33, we cannot reject the null hypothesis (that the betweeness of genres) are different at any reasonable significance level.

Note:  A t test for any of the other metrics can be performed in a similar fashion.